onnxruntime/test/framework/inference_session_test.cc

```diff
@@ -98,8 +98,9 @@ class FuseExecutionProvider : public IExecutionProvider {
   }
 
   std::shared_ptr<::onnxruntime::KernelRegistry> GetKernelRegistry() const override {
-    static std::shared_ptr<::onnxruntime::KernelRegistry>
-        kernel_registry = std::make_shared<::onnxruntime::KernelRegistry>(RegisterOperatorKernels);
+    static std::shared_ptr<KernelRegistry> kernel_registry = std::make_shared<KernelRegistry>();
```

Comment:
If you have a function, that can return a KernelRegistry* directly, you don't need to use call_once at there. The 'static' keyword implies 'call_once'

Reply:
call_once is for calling RegisterOperatorKernels function, instead of creating kernel_registry share pointer.

Reply:
You could create a wrapper function like:
```c++
KernelRegistry* RegisterOperatorKernelsWrapper(){
    KernelRegistry* ret = new KernelRegistry();
    RegisterOperatorKernels(*ret);
    return ret;
}

std::shared_ptr<::onnxruntime::KernelRegistry> GetKernelRegistry() const override {
    static std::shared_ptr<::onnxruntime::KernelRegistry>  kernel_registry = std::make_shared<::onnxruntime::KernelRegistry>(RegisterOperatorKernelsWrapper());
    return kernel_registry;
}

Reply:
Good idea @snnn. I'd suggest using a smart pointer in the wrapper function too for exception safety.

Reply:
thank you guys! will follow the suggestion.